### PR TITLE
Add lp_matrix_data

### LINKED ISSUE #3572: Add standard_form_matrix
x-ref https://discourse.julialang.org/t/matrix-vector-representation-of-a-system-of-linear-inequalities/106188/7?u=odow

### PR DESCRIPTION
**Pull request by odow**
Closes #3572 

TODO:

 - [x] documentation
 - [x] bike shed the name
 - [x] check `lp_sensitivity` tests

### CODE REVIEW

File: test/test_standard_form_matrix.jl [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    @constraint(model, (2 // 3)x + 3y <= 6 // 5)
+    @constraint(model, -4y >= 5z + 7 // 8)
+    @constraint(model, -1 <= x + y <= 2)
+    @objective(model, Min, (1 // 2) - 2x + (1 // 3) * z)
+    a = StandardFormMatrix(model)
+    @test Matrix(a.A) == [1 0 0; 0 -4 -5; (2//3) 3 0; 1 1 0]
+    @test a.b_lower == [5, 7 // 8, -1 // 0, -1]
+    @test a.b_upper == [5, 1 // 0, 6 // 5, 2]
+    @test a.x_lower == [1, 2, 3]
+    @test a.x_upper == [1 // 0, 1//0, 4]
```
Comment (odow) [inline test/test_standard_form_matrix.jl:53]:
```suggestion
    @test a.x_upper == [1 // 0, 1 // 0, 4]
```

File: src/standard_form_matrix.jl [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    sense::MOI.OptimizationSense
+    variable_to_column::Dict{GenericVariableRef{T},Int}
+    affine_constraints::Vector{ConstraintRef}
+    variable_constraints::Vector{ConstraintRef}
+end
+
+"""
+    StandardFormMatrix(model::GenericModel{T})
+
+Given a problem in the form:
```
Comment (jd-foster) [inline src/standard_form_matrix.jl:23]:
```suggestion
Given a JuMP model of any linear program, return a struct storing data for an equivalent LP problem in the form of:
```

**Comment by jd-foster** [comment]
StandardFormData ?

**Comment by odow** [comment]
Other options:

 * `lp_standard_form(model)`
 * `lp_matrix_data(model)`
 * `standard_matrix_data(model)`
 * `standard_matrix_form(model)`
 * `standard_form_matrix(model)`
 * `standard_form_matrix_data(model)`

I think I like `lp_matrix_data`. `LPMatrixData` is a bit ugly, but okay for the name of a struct.

File: src/lp_matrix_data.jl [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    sense::MOI.OptimizationSense
+    variable_to_column::Dict{GenericVariableRef{T},Int}
+    affine_constraints::Vector{ConstraintRef}
+    variable_constraints::Vector{ConstraintRef}
+end
+
+"""
+    lp_matrix_data(model::GenericModel{T})
+
+Given a JuMP model of a linear program, return a [`LPMatrixData{T}`](@ref)
```
Comment (jd-foster) [inline src/lp_matrix_data.jl:29]:
```suggestion
Given a JuMP model of a linear program, return an [`LPMatrixData{T}`](@ref)
```

**Comment by jd-foster** [comment]
I think the name works well.

**Comment by odow** [comment]
@blegat, I think I can anticipate your comment: why not `Utilities.MatrixOfConstraints`. 

I think we potentially could in future, but this is a simple first pass. I'm more interested in your thoughts on the user-facing data structure. There's an argument for providing a matrix and a vector of sets, or `A <sense> b`, but I think we should target what 99% of people want, which is the matrix and vectors from an LP.

File: src/lp_matrix_data.jl [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+struct LPMatrixData{T}
+    A::SparseArrays.SparseMatrixCSC{T,Int}
+    b_lower::Vector{T}
+    b_upper::Vector{T}
+    x_lower::Vector{T}
+    x_upper::Vector{T}
+    c::Vector{T}
+    c_offset::T
+    sense::MOI.OptimizationSense
+    variable_to_column::Dict{GenericVariableRef{T},Int}
```
Comment (blegat) [inline src/lp_matrix_data.jl:21]: Isn't it a bit inconsistent to have a column -> `VariableRef` map but a row -> `ConstraintRef` map ?

Comment (odow) [reply src/lp_matrix_data.jl:21]: That's fair. Although at that point we could just direct people to `all_variables(model)`.

Comment (odow) [reply src/lp_matrix_data.jl]: Updated

**Review by blegat — APPROVED** [review]

**Pull request merged**